core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -86,182 +95,224 @@ public PendingMessage(InputStream message, boolean shouldBeCompressed) {
     }
   }
 
-  DelayedStream(ClientStreamListener listener) {
-    this.listener = listener;
+  @Override
+  public void start(ClientStreamListener listener) {
+    synchronized (this) {
+      // start may be called at most once.
+      checkState(this.listener == null, "already started");
+      this.listener = checkNotNull(listener, "listener");
+
+      // Check error first rather than success.
+      if (error != null) {
+        listener.closed(error, new Metadata());
+      }
+      // In the event that an error happened, realStream will be a noop stream.  We still call
```

Comment:
I left it as unstarted to contrast with the startedStream.  I felt that startedStream might be confusing with realStream